src/aws/osml/model_runner/api/image_request.py

```diff
@@ -21,148 +22,242 @@
+        """
+        properties = ImageRequest._parse_external_message(image_request)
+        return ImageRequest(properties)
+
+    @staticmethod
+    def _parse_external_message(image_request: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Parses an external message dictionary to extract relevant properties.
 
-        if "imageProcessorTileFormat" in image_request:
-            properties["tile_format"] = image_request["imageProcessorTileFormat"]
+        :param image_request: Dictionary containing external message attributes.
+        :return: A dictionary of parsed properties.
+        """
+        properties: Dict[str, Any] = {
+            "job_id": image_request.get("jobId", ""),
+            "image_url": image_request.get("imageUrls", [""])[0],
+            "image_id": f"{image_request.get('jobId', '')}:{image_request.get('imageUrls', [''])[0]}",
+            "image_read_role": image_request.get("imageReadRole", ""),
+            "model_name": image_request["imageProcessor"]["name"],
+            "model_invoke_mode": image_request["imageProcessor"]["type"],
+            "model_invocation_role": image_request["imageProcessor"].get("assumedRole", ""),
+            "tile_size": ImageRequest._parse_tile_dimension(image_request.get("imageProcessorTileSize")),
+            "tile_overlap": ImageRequest._parse_tile_dimension(image_request.get("imageProcessorTileOverlap")),
+            "tile_format": image_request.get("imageProcessorTileFormat", ImageFormats.NITF),
+            "tile_compression": image_request.get("imageProcessorTileCompression", ImageCompression.NONE),
+            "roi": ImageRequest._parse_roi(image_request.get("regionOfInterest")),
+            "outputs": ImageRequest._parse_outputs(image_request),
+            "feature_properties": image_request.get("featureProperties", []),
+            "post_processing": ImageRequest._parse_post_processing(image_request.get("postProcessing")),
+        }
+        return properties
 
-        if "imageProcessorTileCompression" in image_request:
-            properties["tile_compression"] = image_request["imageProcessorTileCompression"]
+    @staticmethod
+    def _parse_tile_dimension(value: Optional[str]) -> ImageDimensions:
+        """
+        Converts a string value to a tuple of integers representing tile dimensions.
 
-        properties["job_id"] = image_request["jobId"]
+        :param value: String value representing tile dimension.
+        :return: Tuple of integers as tile dimensions.
+        """
+        return (int(value), int(value)) if value else DEFAULT_TILE_SIZE
 
-        properties["image_url"] = image_request["imageUrls"][0]
-        properties["image_id"] = image_request["jobId"] + ":" + properties["image_url"]
-        if "imageReadRole" in image_request:
-            properties["image_read_role"] = image_request["imageReadRole"]
+    @staticmethod
+    def _parse_roi(roi: Optional[str]) -> Optional[BaseGeometry]:
```

Comment:
Do we need a whole method for a 1-line shapely loads ternary?  Seems overly verbose.

Reply:
We spoke about these offline and are leaving them to provide readability to the parse_external_message() - logic.